bindings/pyroot/pythonizations/python/ROOT/__init__.py

```diff
@@ -29,58 +29,155 @@
 import warnings
 
 major, minor = sys.version_info[0:2]
-librootpyz_mod_name = 'libROOTPythonizations{}_{}'.format(major, minor)
+librootpyz_mod_name = "libROOTPythonizations{}_{}".format(major, minor)
 importlib.import_module(librootpyz_mod_name)
 
 # ensure 'import libROOTPythonizations' will find the versioned module
-sys.modules['libROOTPythonizations'] = sys.modules[librootpyz_mod_name]
+sys.modules["libROOTPythonizations"] = sys.modules[librootpyz_mod_name]
 
 # Trigger the addition of the pythonizations
 from ._pythonization import _register_pythonizations
+
 _register_pythonizations()
 
 # Check if we are in the IPython shell
 if major == 3:
     import builtins
 else:
-    import __builtin__ as builtins  
+    import __builtin__ as builtins
 
-_is_ipython = hasattr(builtins, '__IPYTHON__')
+_is_ipython = hasattr(builtins, "__IPYTHON__")
 
 # Configure ROOT facade module
 import sys
 from ._facade import ROOTFacade
-sys.modules[__name__] = ROOTFacade(sys.modules[__name__], _is_ipython)
+
+_root_facade = ROOTFacade(sys.modules[__name__], _is_ipython)
+sys.modules[__name__] = _root_facade
+
+# Configure meta-path finder for ROOT namespaces, following the Python
+# documentation and an example:
+#
+#   * https://docs.python.org/3/library/importlib.html#module-importlib.abc
+#
+#   * https://python.plainenglish.io/metapathfinders-or-how-to-change-python-import-behavior-a1cf3b5a13ec
+from importlib.abc import Loader, MetaPathFinder
+from importlib.machinery import ModuleSpec
+from importlib.util import spec_from_loader
+
+
+def _can_be_module(obj) -> bool:
+    """
+    Determine if an object can be used as a Python module. This is the case for
+    objects that are actually of ModuleType, or C++ namespaces from cppyy.
+    """
+
+    # If the type is the module type, it can trivially be a module.
+    if isinstance(obj, types.ModuleType):
+        return True
+
+    # Check if the object represents a C++ namespace. Since cppyy has no
+    # dedicated Python type for C++ namespaces, we check for this using the
+    # representation of the object.
+    if repr(obj).startswith("<namespace "):
```

Comment:
Does this pass through the pretty printer of the interpreter? (I think no, but triple checking b/c that would be a perf hit)

Reply:
Good question, I will investigate

Reply:
I found where it's implemented. `__repr__` doesn't call the interpreter, but CPyCppyy builds the representation from some internal flags:
https://github.com/root-project/root/blob/master/bindings/pyroot/cppyy/CPyCppyy/src/CPPScope.cxx#L175

Reply:
Both classes and namespaces are instances of `cppyy.types.Scope` as there's no Python reason to distinguish them. For C++ reasons to differentiate, `cppyy.reflex` exists. E.g. `obj.__cpp_reflex__(cppyy.reflex.IS_NAMESPACE)` should give the right answer. (This all in master anyway.)

Reply:
Thanks a lot! I'll keep this in mind for the next changes to the ROOT module.